File: Solutions/0448- Find All Numbers Disappeared in an Array/Solution.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        for i in range(0,len(nums)):
            while nums[nums[i] - 1] != nums[i]:
                nums[nums[i] - 1], nums[i] = nums[i], nums[nums[i] - 1]
        
        answer = list()

        for i in range(0,len(nums)):
            if nums[i] != i + 1:
                answer.append(i + 1)

        return answer
        

File: Solutions/0448- Find All Numbers Disappeared in an Array/Solution.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        for (int i = 0; i < nums.length; i++) {
            while (nums[nums[i] - 1] != nums[i]) {
                int temp = nums[nums[i] - 1];
                nums[nums[i] - 1] = nums[i];
                nums[i] = temp;
            }
        }

        List<Integer> answer = new ArrayList<>();

        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != i+1) {
                answer.add(i+1);
            }
        }
        
        return answer;
    }
}